src/lib/ActionPrototype.js

```diff
@@ -1,29 +1,57 @@
-var fetcher = require('../lib/jsonFetcher');
-var AppDispatcher = require('../lib/ep-dispatcher');
-var KeyMirror = require('keymirror');
+import fetcher from '../lib/jsonFetcher'
+import AppDispatcher  from '../lib/ep-dispatcher'
+import KeyMirror from 'keymirror'
+
+//TODOS: 
+//--turn into a class. 
+//--make it a factory function like we do with store prototype. 
 
 var ActionPrototype = {
   specialTypes: KeyMirror({
     GOT_MSG: null
   }),
+  originalActions: {},
+  stubForTests: function() {
+    this.recievedData = {}
+    let actions = Object.getOwnPropertyNames(this).filter( (fn)=> {return typeof this[fn] === 'function' } )
```

Comment:
gah... okay. I can be talked out of this. For some reason my eye gets jumbled at the `((arg) => { })`

I like the `()=>` as one nugget. But Happy to differ.